src/Datagrid/Pager.php

```diff
@@ -42,7 +42,7 @@ public function computeNbResult()
             current($this->getCountColumn())
         ));
 
-        return $countQuery->resetDQLPart('orderBy')->getQuery()->getSingleScalarResult();
+        return $countQuery->resetDQLPart('orderBy')->resetDQLPart('groupBy')->getQuery()->getSingleScalarResult();
```

Comment:
Ok so let's admit this provides a better result in case of empty set. What about other cases? Doesn't it provide a wrong result because there's no grouping and you are counting rows with several relationships more than once?

Reply:
Thank you for reviewing @greg0ire !

Actually, without removing the groupBy part it provides wrong results (count always 1). I guess, because it's overriding the select part `->select("NEW NS\AdminBundle..`

With my current data set, the counting works well with empty, filtered or not filtered data.

Reply:
are you sure? what if you have 5 configurations for one interface?

Reply:
pretty sure, it's grouping by the interface id.. But I see your point, that it might not work for all cases. I'll will check for another solution. Thank you!